### ISSUE TITLE
Crash: HHVM 3.5.0 segfaults duing mysql_get_server_info

### ISSUE THREAD

**Issue by tengyifei** [facebook/hhvm#4974]
HHVM 3.5.0 segfaults during what looks like a call to `mysql_get_server_info`.
```
Version: tags/HHVM-3.5.0-0-ga90f4733cfa0d8fefdafc7433f758f78cdc75424

Stack trace:
0 __restore_rt at sigaction.c:0
1 HPHP::MySQL::GetConn(HPHP::Variant const&, std::shared_ptrHPHP::MySQL) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
2 HPHP::f_mysql_get_server_info(HPHP::Variant const&) at ext_mysql.cpp:0
3 void HPHP::Native::callFunc(HPHP::Func const, void, HPHP::TypedValue, HPHP::TypedValue&) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
4 void HPHP::ExecutionContext::dispatchImpl() at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
5 HPHP::ExecutionContext::enterVM(HPHP::ActRec, HPHP::ExecutionContext::StackArgsState, HPHP::Resumable, HPHP::ObjectData) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
6 HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue, HPHP::Func const, HPHP::Variant const&, HPHP::ObjectData, HPHP::Class, HPHP::VarEnv, HPHP::StringData, HPHP::ExecutionContext::InvokeFlags) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
7 HPHP::ExecutionContext::invokeUnit(HPHP::TypedValue, HPHP::Unit const) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
8 HPHP::include_impl_invoke(HPHP::String const&, bool, char const) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
9 HPHP::hphp_invoke(HPHP::ExecutionContext, std::basic_string, std::allocator > const&, bool, HPHP::Array const&, HPHP::VRefParamValue const&, std::basic_string, std::allocator > const&, std::basic_string, std::allocator > const&, bool&, std::basic_string, std::allocator >&, bool, bool, bool) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
10 HPHP::HttpRequestHandler::executePHPRequest(HPHP::Transport, HPHP::RequestURI&, HPHP::SourceRootInfo&, bool) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
11 HPHP::HttpRequestHandler::handleRequest(HPHP::Transport) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
12 HPHP::ServerWorkerstd::shared_ptr<HPHP::FastCGIJob, HPHP::FastCGITransportTraits>::doJobImpl(std::shared_ptrHPHP::FastCGIJob, bool) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
13 HPHP::JobQueueWorkerstd::shared_ptr<HPHP::FastCGIJob, HPHP::Server, true, false, HPHP::JobQueueDropVMStack>::start() at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
14 HPHP::AsyncFuncImpl::ThreadFunc(void) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
15 HPHP::start_routine_wrapper(void) at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/hhvm:0
16 start_thread at pthread_create.c:0
17 __clone at /var/lib/openshift/54fa65a3e0b8cd5c720000a8/nginx-hhvm/usr/bin/libc.so.6:0
```

This is basically a mirror of tengyifei/openshift-cartridge-nginx-hhvm#14

**Comment by paulbiss** [comment]
Do you see this crash on master?

**Comment by tengyifei** [comment]
As you can see this crash originated from a user of my custom packaged cartridge, which is at 3.5.0-rel. I'll do a compile from master and tell him to give it a try.

**Comment by puppywang** [comment]
I've see the same crash using prebuilt version on ubuntu 14.04 x64, here is the dump file.

Host: iZ25k5gs60vZ
ProcessID: 14154
ThreadID: 7f2370fff700
ThreadPID: 14157
Name: unknown program
Type: Segmentation fault
Runtime: hhvm
Version: tags/HHVM-3.5.1-0-gb723e67c88bc008676687a04dff00db1faf250b3
DebuggerCount: 0

ThreadType: Web Request
Server_SERVER_NAME: dajia365.com
Server: www.dajia365.com:8180
URL: /index.php/Admin/Main/index.html

# 0  HPHP::MySQL::GetConn(HPHP::Variant const&, std::shared_ptr<HPHP::MySQL>*) at /usr/bin/hhvm:0
# 1  HPHP::f_mysql_get_server_info(HPHP::Variant const&) at ext_mysql.cpp:0
# 2  void HPHP::Native::callFunc<false, false>(HPHP::Func const*, void*, HPHP::TypedValue*, HPHP::TypedValue&) at /usr/bin/hhvm:0
# 3  HPHP::TypedValue* HPHP::Native::functionWrapper<false, false>(HPHP::ActRec*) at /usr/bin/hhvm:0
# 4  HPHP::jit::x64::BackEnd::enterTCHelper(unsigned char*, HPHP::jit::TReqInfo&) at /usr/bin/hhvm:0
# 5  HPHP::jit::MCGenerator::enterTC(unsigned char*, void*) at /usr/bin/hhvm:0
# 6  HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, HPHP::Func const*, HPHP::Variant const&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags) at /usr/bin/hhvm:0
# 7  HPHP::ObjectData::o_invoke(HPHP::String const&, HPHP::Variant const&, bool) at /usr/bin/hhvm:0
# 8  HPHP::f_hphp_invoke_method(HPHP::Variant const&, HPHP::String const&, HPHP::String const&, HPHP::Variant const&) at /usr/bin/hhvm:0
# 9  HPHP::jit::x64::BackEnd::enterTCHelper(unsigned char*, HPHP::jit::TReqInfo&) at /usr/bin/hhvm:0
# 10 HPHP::jit::MCGenerator::enterTC(unsigned char*, void*) at /usr/bin/hhvm:0
# 11 HPHP::ExecutionContext::invokeFunc(HPHP::TypedValue*, HPHP::Func const*, HPHP::Variant const&, HPHP::ObjectData*, HPHP::Class*, HPHP::VarEnv*, HPHP::StringData*, HPHP::ExecutionContext::InvokeFlags) at /usr/bin/hhvm:0
# 12 HPHP::ExecutionContext::invokeUnit(HPHP::TypedValue*, HPHP::Unit const*) at /usr/bin/hhvm:0
# 13 HPHP::invoke_file(HPHP::String const&, bool, char const*) at builtin-functions.cpp:0
# 14 HPHP::include_impl_invoke(HPHP::String const&, bool, char const*) at /usr/bin/hhvm:0
# 15 HPHP::hphp_invoke(HPHP::ExecutionContext*, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, HPHP::Array const&, HPHP::VRefParamValue const&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool&, std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool, bool, bool) at /usr/bin/hhvm:0
# 16 HPHP::HttpRequestHandler::executePHPRequest(HPHP::Transport*, HPHP::RequestURI&, HPHP::SourceRootInfo&, bool) at /usr/bin/hhvm:0
# 17 HPHP::HttpRequestHandler::handleRequest(HPHP::Transport*) at /usr/bin/hhvm:0
# 18 HPHP::ServerWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::FastCGITransportTraits>::doJob(std::shared_ptr<HPHP::FastCGIJob>) at /usr/bin/hhvm:0
# 19 HPHP::JobQueueWorker<std::shared_ptr<HPHP::FastCGIJob>, HPHP::Server*, true, false, HPHP::JobQueueDropVMStack>::start() at /usr/bin/hhvm:0
# 20 HPHP::AsyncFuncImpl::ThreadFunc(void*) at /usr/bin/hhvm:0
# 21 HPHP::start_routine_wrapper(void*) at /usr/bin/hhvm:0
# 22 start_thread at /lib/x86_64-linux-gnu/libpthread.so.0:0
# 23 clone at /lib/x86_64-linux-gnu/libc.so.6:0

PHP Stacktrace:

#0  mysql_get_server_info() called at [/usr/share/nginx/html/application/Admin/Controller/MainController.class.php

**Comment by fredemmott** [comment]
This appears to be fixed in nightlies and 3.6:

http://3v4l.org/uYcB3 - exit code 139 is SIGSEGV, but the same code works fine in 3.6.

Can you confirm that 3.6 fixes or does not fix your problem?

**Comment by puppywang** [comment]
Confirmed,  no more crash, thank you
